src/utils/api/helper.ts

```diff
@@ -1161,3 +1162,44 @@ export const recaptchaTransaction = async ({
     onErrorActivate: onError,
   });
 };
+
+export const evaluateFeatureFlag = async (
+  featureKey: string,
+  onError: (e: string) => void,
+  onResponse: (data: { enabled: boolean }) => void
+) => {
+  await pipe(
+    TE.tryCatch(
+      () => apiCheckoutFeatureFlags.evaluateFeatureFlag({ featureKey }),
+      () => {
+        onError("Network error");
+        return new Error("Network error");
+      }
+    ),
+    TE.fold(
+      () => async () => {
+        onError("Server error");
+        return {};
+      },
+      (response) => async () =>
+        pipe(
+          response,
+          E.fold(
+            () => {
+              onError("Response error");
```

Comment:
What happens if the API is unavailable? What is the default value of the feature flag?